services/opsgenie/service.go

```diff
@@ -0,0 +1,125 @@
+package opsgenie
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"log"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/influxdb/kapacitor"
+)
+
+type Service struct {
+	apikey       string
+	teams        string
+	recipients   string
+	url          string
+	recovery_url string
+	global       bool
+	logger       *log.Logger
+}
+
+func NewService(c Config, l *log.Logger) *Service {
+	return &Service{
+		teams:        c.Teams,
+		recipients:   c.Recipients,
+		apikey:       c.APIKey,
+		url:          c.URL + "/",
+		recovery_url: c.RecoveryURL + "/",
+		global:       c.Global,
+		logger:       l,
+	}
+}
+
+func (s *Service) Open() error {
+	return nil
+}
+
+func (s *Service) Close() error {
+	return nil
+}
+
+func (s *Service) Global() bool {
+	return s.global
+}
+
+func (s *Service) Alert(teams, recipients, messageType, message, entityID string, t time.Time, details interface{}) error {
+	ogData := make(map[string]interface{})
+	override := ""
+	url := s.url
+
+	ogData["apiKey"] = s.apikey
+	ogData["entity"] = entityID
+	ogData["alias"] = entityID
+	ogData["message"] = message
+	ogData["note"] = ""
+	ogData["monitoring_tool"] = kapacitor.Product
+
+	//Details
+	ogDetails := make(map[string]interface{})
+	ogDetails["Level"] = messageType
+	ogDetails["Monitoring Tool"] = "Kapacitor"
+
+	ogData["details"] = ogDetails
+
+	switch messageType {
+	case "CRITICAL":
+		override = "OverwritesQuietHours"
+	case "RECOVERY":
+		url = s.recovery_url
+		ogData["note"] = message
+	}
+
+	ogData["tags"] = []string{override, messageType}
+
+	if details != nil {
+		b, err := json.Marshal(details)
+		if err != nil {
+			return err
+		}
+		ogData["description"] = string(b)
+	}
+
+	if teams == "" {
+		teams = s.teams
+	}
+
+	if recipients == "" {
+		recipients = s.recipients
+	}
+
+	if teams != "" {
+		ogData["teams"] = strings.Split(teams, ",")
```

Comment:
Can we make the `teams` and `recipients` vars a list of strings instead of splitting on commas?